lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -196,14 +196,14 @@ def parse_relationship(assoc_name, assoc_data, options)
         # @api private
         def parse_relationships(relationships, options)
           transform_keys(relationships, options)
-            .map { |(k, v)| parse_relationship(k, v[:data], options) }
+            .map { |(k, v)| parse_relationship(k, v['data'], options) }
```

Comment:
Not sure if it's a bug. KeyTransform in master always returned symbols so a recent PR changes this code to work with that restriction. This PR makes Transform more flexible so that set of changes isn't necessary anymore.